### ISSUE TITLE
Use Yglu from code

### ISSUE THREAD

**Issue by vladdu** [lbovet/yglu#7]
Hi! Currently, yglu is meant to be use as a command-line tools, but I would need to be able to embed it in other code, so that I can process the results further, and also be able to create the input dynamically (possibly using non-yaml data. Would it be interesting to contribute back these changes? BTW, do you plan to maintain the tool?

**Comment by lbovet** [comment]
You mean changes in Yglu so that it has a clean API for being embedded? Yes it definitely makes sense.
About maintenance, I am not planning new features but would happily fix bugs and welcome PRs.

**Comment by vladdu** [comment]
Yes, that's what I was thinking. I will check with my team if we think we want to use yglu and make it work for us, and I will probably bother you with questions once in a while.

**Comment by lbovet** [comment]
Good for me. A starting point would be to make the builder accept a yaml structure instead of the yaml text. Then, one can use it from application code, as `main.py` does.

**Comment by vladdu** [comment]
I sent a PR for a simple way to do this. 

If I run pylint on the code base, there are a lot of warnings. If I fix those, will you accept them (even if they might be more than cosmetic)? I mean for example the usage of `errors=[]` as parameters, which can give unexpected results (see https://florimond.dev/blog/articles/2018/08/python-mutable-defaults-are-the-source-of-all-evil/)

Also, what formatter did you use? I usually use yapf with its defaults, but that would give many diffs with your code.

Finally, the repo doesn't have any venv setup, so it's a bit tricky to set up. Should I add it?

**Comment by lbovet** [comment]
Oh, what a nasty thing these mutable defaults. I did not use a particular formatter, I think I just relied on the defaults of vscode.
Yes, formatting the codebase is absolutely ok for me. Ideally, along with an .editorrc or something that makes editors aware of the correct formatting. And linting in the build process.
Also gratefully accepting PR for warning removals. Ideally separate commits for the formatting and fixes.
If venv is the way to go for easing the development, yes let's add it!
Overall, I am very glad to welcome experienced python devs on this project!
